File: api/src/main/java/org/bigcraft/lootpool/api/LootPoolApi.java
package org.bigcraft.lootpool.api;

public final class LootPoolApi {
    private static LootPoolProvider provider;

    public static void setProvider(LootPoolProvider provider) {
        LootPoolApi.provider = provider;
    }

    public static LootPoolProvider getProvider() {
        return provider;
    }
}

File: src/main/kotlin/me/wyne/lootpool/LootPoolApi.kt
package me.wyne.lootpool

import com.google.inject.Inject
import com.google.inject.Singleton
import me.wyne.lootpool.api.LootPoolApi
import me.wyne.lootpool.api.LootPoolProvider
import me.wyne.lootpool.core.LootPoolManager
import org.bukkit.plugin.ServicePriority

@Singleton
class LootPoolApi @Inject constructor(plugin: LootPool, lootPoolManager: LootPoolManager) {
    init {
        plugin.server.servicesManager.register(LootPoolProvider::class.java, lootPoolManager, plugin, ServicePriority.Normal)
        LootPoolApi.setProvider(lootPoolManager)
    }
}